pennylane/transforms/decompose_single_qubit_unitaries.py

```diff
@@ -0,0 +1,86 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+A transform for decomposing arbitrary single-qubit QubitUnitary gates into elementary gates.
+"""
+import pennylane as qml
+from pennylane import math
+from pennylane.transforms import qfunc_transform
+from pennylane.transforms.decompositions import zyz_decomposition
+
+
+@qfunc_transform
+def decompose_single_qubit_unitaries(tape):
```

Comment:
> decompose_single_qubit_unitaries(tape, decomp=zyz_decomposition)

Oh I like this idea!

> A separate thought: I don't really like the name, but am going to be really annoying when I say I don't have any better suggestions 
> 
> I find it a bit too long
> I keep remembering that the compile PR uses 'expand' instead of 'decompose'

Yes I feel the same (about this and some of the other compilation transforms). Some other ideas:
 - `decompose_qubit_unitaries`
 - `unitaries_to_rot`  / `unitary_to_rot` 
 - `qubit_unitary_to_rot`

Reply:
Oh I like `unitary_to_rot`

Reply:
Me too :+1: It's the shortest and also captures that it's for single-qubit operations :tada:

Reply:
> decompose_single_qubit_unitaries(tape, decomp=zyz_decomposition)

Actually, if we are going to have the transform convert things to `Rot`, it makes sense to keep as-is rather than adding the option, since `Rot` is intrinsically `zyz`. But later if we add more decompositions and start returning them as `RZ, RY, RZ`, etc. instead of `Rot`, this would make things more flexible.